### PR TITLE
feat: [PL-32491]: OpenAPI Spec changes for Entity favourites.

### PR DESCRIPTION
**Pull request by boopesh007**
## Describe your changes
 OpenAPI Spec changes for Entity favourites.

## Checklist
- [x] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46986)

### CODE REVIEW

**Comment by boopesh007** [comment]
trigger ut0

File: 120-ng-manager/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -1090,6 +1099,161 @@ paths:
...
+      responses:
+        '200':
+          $ref: '#/components/responses/FavouriteListResponse'
+      tags:
+        - Account Favourites
+    delete:
+      summary: Delete a favourite
+      operationId: delete-account-favourite
+      responses:
+        '200':
```
Comment (mankrit-singh) [inline 120-ng-manager/contracts/openapi/v1/openapi.yaml:1140]: can make this 204 (No content) according to standards

File: 120-ng-manager/contracts/openapi/v1/openapi.yaml [hunk 2]
```diff
@@ -1090,6 +1099,161 @@ paths:
...
+      description: Creates favourite resource of a user at account level
+      parameters:
+        - $ref: '#/components/parameters/AccountHeader'
+        - $ref: '#/components/parameters/UserId'
+        - $ref: '#/components/parameters/ResourceType'
+        - $ref: '#/components/parameters/ResourceId'
+      security:
+        - x-api-key: []
+      responses:
+        '200':
```
Comment (mankrit-singh) [inline 120-ng-manager/contracts/openapi/v1/openapi.yaml:1115]: if we are creating an entity, this should be 201

**Review by mankrit-singh — COMMENTED** [review]
also please mark these with x-internal:true if these are to be merged without implementation. thanks!

File: 120-ng-manager/contracts/openapi/v1/openapi.yaml [hunk 3]
```diff
@@ -1175,6 +1339,82 @@ components:
...
+          pattern: '^[a-zA-Z_][0-9a-zA-Z_$]{0,127}$'
+          minLength: 1
+          maxLength: 128
+        module:
+          type: string
+          description: Harness module
+          pattern: '^[a-zA-Z_][0-9a-zA-Z_$]{0,127}$'
+          minLength: 1
+          maxLength: 128
+        resourceType:
```
Comment (mankrit-singh) [inline 120-ng-manager/contracts/openapi/v1/openapi.yaml:1386]: here also it should be snake case: for user_id, resource_type and resource_id

**Review by gn-harness — APPROVED** [review]

**Review by mankrit-singh — APPROVED** [review]

**Comment by boopesh007** [comment]
trigger messagecheck

**Comment by boopesh007** [comment]
trigger messagecheck

**Review by mankrit-singh — APPROVED** [review]

**Pull request merged**